Connection banner: display banner inside React page.

jeherve opened the pull request:
Fixes #11071

#### Changes proposed in this Pull Request:

- The fullscreen connection banner appearing on Jetpack > Dashboard previously overlaid all other information on the page. This included useful information like footer links, but also potential error messages.
- This PR adds a new empty div to the main react admin page structure when a site is not connected to WordPress.com, and then moves the content from the notice that contains the connection banner to inside that empty div, in the middle of the page.

This creates a difference between the fullscreen connection banner in the plugins menu and the one in Jetpack > Dashboard: since the Jetpack dashboard already includes a masthead with a Jetpack logo, I removed the logo from the connection banner there.

#### Screenshots

**Plugins screen**

<img width="1461" alt="screenshot 2019-01-08 at 21 16 24" src="https://user-images.githubusercontent.com/426388/50856529-41648780-138b-11e9-8af6-cd1a318bedaa.png">

**Jetpack > Dashboard before**

<img width="1464" alt="screenshot 2019-01-08 at 21 21 12" src="https://user-images.githubusercontent.com/426388/50856582-5f31ec80-138b-11e9-86a1-da8e542fbc2b.png">

**Jetpack > Dashboard after**

<img width="1466" alt="screenshot 2019-01-08 at 21 16 47" src="https://user-images.githubusercontent.com/426388/50856544-4b868600-138b-11e9-9b7f-80d60ae89609.png">

### Testing instructions:

1. Deactivate the Jetpack plugin on a site running this branch.
2. Activate the plugin again.
3. View the fullscreen connection banner under the Plugins menu
4. Go to Jetpack > Dashboard and view the new fullscreen connection banner. Make sure it looks good in all browsers.
5. Add some random characters before the opening `<?php` tag in your site's `wp-config.php` file
6. Attempt to connect the site to WordPress.com.
7. Notice that you can see the error message when the connection fails.

#### Proposed changelog entry for your changes:

None.

jeherve committed:
Connection banner: display banner inside React page.

Fixes #11071

- The fullscreen connnection banner appearing on Jetpack > Dashboard previously overlaid all other
information on the page. This included useful information like footer links, but also potential error
messages.
- This PR adds a new empty div to the main react admin page structure when a site is not connected to
WordPress.com, and then moves the content from the notice that contains the connection banner to inside
that empty div, in the middle of the page.

This creates a difference between the fullscreen connection banner in the plugins menu and the one in
Jetpack > Dashboard: since the Jetpack dashboard already includes a masthead with a Jetpaack logo, I
removed the logo from the connection banner there.

dereksmart commented:
I think we can probably remove/hide this card? 

![image](https://user-images.githubusercontent.com/7129409/50858865-b326dc00-135f-11e9-97b1-b70ac9064e91.png)

jeherve committed:
Remove support card from admin page when the site is not connected

See https://github.com/Automattic/jetpack/pull/11105#issuecomment-452452055

jeherve commented:
@dereksmart Good call. Now that we mention support in the banner, no need to mention it again below. I restricted the card to connected sites in 1166c3e51c4bdf7c6e2475f07af6652b3cf332f1

joanrho commented:
@jeherve I can see the banner on the disconnecting plugins screen view, but not in the Jetpack Dashboard view.

jeherve commented:
@joanrho What do you see there instead? Could you make a screenshot?

Could you also try with a brand new website like [this one](https://jurassic.ninja/create?jetpack-beta&branch=update/disconnected-jetpack&wp-debug-log) and let me know how it goes?

rickybanister commented:
@jeherve if you pull this off you've done the impossible! I think I first proposed this in January 2018. Kudos to you!

jeherve commented:
@rickybanister Feel free to give it a try on a test site like [this one](https://jurassic.ninja/create?jetpack-beta&branch=update/disconnected-jetpack&wp-debug-log), I would be curious to see if that works for you, given that it seems buggy for @joanrho 	🤞 	

Thank you!

rickybanister commented:
It  worked just fine for me. I have one minor suggestion, which is to add the `card` border/shadow to the banner in order to mirror the card style elsewhere (like the adjacent apps download card)

![image](https://user-images.githubusercontent.com/437258/50906616-799fb080-13f3-11e9-9c2b-6c93897443c6.png)

Otherwise it worked great.

keoshi reviewed (dismissed):
Looks and works great in my testing!

jeherve committed:
Display connection banner if the connection is set to false

This should solve this issue:
https://github.com/Automattic/jetpack/pull/11105#issuecomment-452507949

jeherve commented:
@joanrho I believe this should solve your issue: 71601e4f8852610a4bfd961ae76bcac6b25912b5

jeherve committed:
Add box shadow around main connection card as well.

See https://github.com/Automattic/jetpack/pull/11105#issuecomment-452721745

jeherve commented:
@rickybanister 91e3c0c6caa1ffaf17e53cd36cb9deb99d39e1c7 should add the shadow:

![image](https://user-images.githubusercontent.com/426388/50911100-71fdfd00-142f-11e9-8919-368d6beb20df.png)

dereksmart reviewed (approved):
LGTM!

joanrho commented:
@jeherve while your link set up a JN site for me, deactivating/reactivating Jetpack seems to have switched me back to the Latest Stable branch, so I was only viewing the old banner on the disconnected Jetpack Dashboard view. I switched back to this PR branch using the “Search for a Jetpack Feature Branch” in the beta plugin, then clicked back to JPDash and then could see the banner updates. (Thanks to @oskosk for helping me troubleshoot that!)

Everything looks great. Thank you for making this work across both screens. Approved!

jeherve commented:
Cherry-picked to `branch-6.9` in 1b2840dc0d0e377f60654f86384a8dbea534dc0f
